internal/memdb/memdb.go

```diff
@@ -205,6 +216,10 @@ func (c *client) Close(context.Context) error {
 
 func (c *client) DeleteStale(ctx context.Context, tables schema.Tables, source string, syncTime time.Time) error {
 	for _, table := range tables {
+		if c.sourceSpec.Backend != specs.BackendNone && table.IsIncremental {
```

Comment:
I'd rather have this assertion in a test than in memdb itself, if possible? The assertion might look different, but it would basically be asserting that an incremental table didn't have all its records cleared after a write

Reply:
I'll try to add this as an assertion then